app/sensors/Boiler.py

```diff
@@ -181,10 +188,34 @@ async def update(self):
                             self.config['preset_mode_state_topic'],
                             set_preset,
                             qos=0, retain=True)
-            if 'thermicLevel' in self.attributes:
+
+            if self.current_setpoint is not None and self.current_temp is not None:
```

Comment:
Also proposition for a tiny refactor here:
Maybe you can have your `if` compute the `hvac_action` and only call the publish once at the end like this:

```python
    self.mqtt.mqtt_client.publish(
        self.config['action_topic'],
        hvac_action,
        qos=1, retain=True)
```

This would make the code more readable IMO.

Reply:
Done